Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -421,6 +421,8 @@ AccessibilityRole AccessibilityNodeObject::determineAccessibilityRoleFromNode(Tr
     // menu tags with toolbar type should have Toolbar role.
     if (node()->hasTagName(menuTag) && equalLettersIgnoringASCIICase(getAttribute(typeAttr), "toolbar"_s))
         return AccessibilityRole::Toolbar;
+    if (node()->hasTagName(menuTag) && !equalLettersIgnoringASCIICase(getAttribute(typeAttr), "toolbar"_s))
```

Comment:
Rather than checking for the menu tagname twice, could we structure it differently to do less work? Maybe something like this?

```
if (node()->hasTagName(menuTag)) {
    if (equalLettersIgnoringASCIICase(getAttribute(typeAttr), "toolbar"_s))
        return AccessibilityRole::Toolbar;
    return AccessibilityRole::List;
}

Reply:
Perfect. I wasn't sure how to structure this. Thanks, @twilco. Let me make an upate

Reply:
Was about to suggest the same thing. This looks good to me now. Although perhaps we want to add a C++ comment about why we are not following the spec here e.g.

    // Per the HTML AAM, <menu> is mapped to list role but we keep backward compatibility for the obsolete type="toolbar".
    // See https://html.spec.whatwg.org/multipage/obsolete.html#attr-menu-type